scripts/compare_warehouse_tables.py

```diff
@@ -0,0 +1,244 @@
+        if prod_table_columns == dev_table_columns:
+            newrows.append("True")
+        else:
+            newrows.append("False")
+            more_info += f"{table_name} table columns Not Matched on " + (", ".join(list(list(set(prod_table_columns) - set(dev_table_columns)) + list(set(dev_table_columns) - set(prod_table_columns))))) + " \n"
+
+        # Adding Rows_Match boolean and percentage (if not 100%) to the newrow row
+        if dev_table_rows is None:
+            newrows.append("True")
+        else:
+            match_percent = int(100 - (((dev_table_rows + prod_table_rows) / dev_table.num_rows) * 100))
+            if match_percent > 85:
+                newrows.append("True (" + str(match_percent) + "%)")
+            else:
+                newrows.append("False")
+                more_info += f"{table_name} table rows Not Matched, only (" + str(match_percent) + "%) the same \n"
+
+        # Adding information about today, yesterday, one month ago, and six month ago matchs to more info
+        if has_timestamp and not_matched:
+            more_info += f"{table_name} table didn't 100% match: " + not_matched_info + " \n"
+        elif has_timestamp:
+            more_info += f"{table_name} table match 100% yesterday, one month, six month \n"
+
+        # Adding In_Dev_Only count to the newrow row
+        newrows.append(dev_table_rows)
+
+        # Adding In_Prod_Only count to the newrow row
+        newrows.append(prod_table_rows)
+
+        # Adding newrow row to the master table list
+        results_newrows.append(newrows)
+
+    # Creating the table of new rows in dev and prod from results_newrows
+    newrows_df = pd.DataFrame(results_newrows, columns=["Table_Name", "Columns_Match?", "Rows_Match?", "In_Dev_Only", "In_Prod_Only"])
+    
+    # Returning the table and information of new rows in dev and prod.
+    return newrows_df, more_info
+
+
+if __name__ == "__main__":
+
+    # Optional Verbose Option
+    parser = argparse.ArgumentParser(description='')
+    parser.add_argument('--verbose', "-v", type=str, default=False, help='Verbose')
+    args = parser.parse_args()
+    verbose = args.verbose
+
+    if not verbose:
+        # Ignore all warning outputs
+        warnings.filterwarnings("ignore")
+
+    # Hard-coded project name for warehouse tables
+    project = "broad-dsde-prod-analytics-dev"
+
+    # Construct a BigQuery client object.
+    bq = bigquery.Client(project)
+
+    # Getting all the warehouse tables
+    query_get_dev_schema = "SELECT * FROM broad-dsde-prod-analytics-dev.warehouse_dev.INFORMATION_SCHEMA.TABLES"
+    dev_dataset_schema = call_bigquery(bq, query_get_dev_schema, project, verbose)
+
+    # Getting all the warehouse tables names
+    dev_table_list = get_tables(dev_dataset_schema, verbose)
+
+    # Dict with the warehouse tables that has a date value 
+    tables_with_timestamps = {"terra_nps": "timestamp",
```

Comment:
it would be nice to pull this programmatically from the schema

Reply:
(those are in workbench-analytics, which is where you'll want to put this script, not in horsefish!)